apps/tai/lib/tai/advisor.ex

```diff
@@ -168,93 +156,38 @@ defmodule Tai.Advisor do
-        key = {venue_id, product_symbol}
-        old_market_quotes = state.market_quotes
-        updated_market_quotes_data = Map.put(old_market_quotes.data, key, current_inside_quote)
-        updated_market_quotes = Map.put(old_market_quotes, :data, updated_market_quotes_data)
-
-        state
-        |> Map.put(:market_quotes, updated_market_quotes)
-      end
-
-      defp inside_quote_is_stale?(
-             previous_inside_quote,
-             %Tai.Markets.OrderBook{bids: bids, asks: asks} = changes
-           ) do
-        (bids |> Enum.any?() && bids |> inside_bid_is_stale?(previous_inside_quote)) ||
-          (asks |> Enum.any?() && asks |> inside_ask_is_stale?(previous_inside_quote))
-      end
-
-      defp inside_bid_is_stale?(_bids, nil), do: true
-
-      defp inside_bid_is_stale?(bids, %Tai.Markets.Quote{} = prev_quote) do
-        bids
-        |> Enum.any?(fn {price, {size, _processed_at, _server_changed_at}} ->
-          price >= prev_quote.bid.price ||
-            (price == prev_quote.bid.price && size != prev_quote.bid.size)
-        end)
-      end
-
-      defp inside_ask_is_stale?(asks, nil), do: true
-
-      defp inside_ask_is_stale?(asks, %Tai.Markets.Quote{} = prev_quote) do
-        asks
-        |> Enum.any?(fn {price, {size, _processed_at, _server_changed_at}} ->
-          price <= prev_quote.ask.price ||
-            (price == prev_quote.ask.price && size != prev_quote.ask.size)
-        end)
-      end
-
       defp execute_handle_inside_quote(
-             state,
+             {data, state},
              venue_id,
-             product_symbol,
-             changes,
-             previous_inside_quote \\ nil
+             product_symbol
            ) do
-        current_inside_quote =
-          state.market_quotes |> Tai.Advisors.MarketQuotes.for(venue_id, product_symbol)
-
-        if current_inside_quote == previous_inside_quote do
-          state
-        else
-          try do
-            with {:ok, new_store} <-
-                   handle_inside_quote(
-                     venue_id,
-                     product_symbol,
-                     current_inside_quote,
-                     changes,
-                     state
-                   ) do
-              Map.put(state, :store, new_store)
-            else
-              unhandled ->
-                Tai.Events.info(%Tai.Events.AdvisorHandleInsideQuoteInvalidReturn{
-                  advisor_id: state.advisor_id,
-                  group_id: state.group_id,
-                  venue_id: venue_id,
-                  product_symbol: product_symbol,
-                  return_value: unhandled
-                })
-
-                state
-            end
-          rescue
-            e ->
-              Tai.Events.info(%Tai.Events.AdvisorHandleInsideQuoteError{
+        try do
+          with {:ok, new_store} <- handle_inside_quote(venue_id, product_symbol, data, state) do
```

Comment:
Does this execute `handle_inside_quote` when the quote hasn't changed?

Reply:
Yes, this gets called when new changes are available.

Reply:
This is a change in behavior. I previously had a callback called `handle_changes` I think, a long time ago but removed it because I wasn't using it. I feel like it would be better to add that back and keep the current behavior for `handle_inside_quote`.

I'd like to call the new callback `handle_event` instead of `handle_changes` so that it can be used for more than order book events.